test/integration/apiserver/cel/mutatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,925 @@
+					ReinvocationPolicy: ptr.To(v1alpha1.NeverReinvocationPolicy),
+					Expression: `
+						Object{
+							metadata: Object.metadata{
+								annotations: {
+									"foo1": "foo1Value"
+								}
+							}
+						}`,
+				}),
+				policy("policy-2", matchAllResources, nil, v1alpha1.Mutation{
+					PatchType:          ptr.To(v1alpha1.ApplyConfigurationPatchType),
+					ReinvocationPolicy: ptr.To(v1alpha1.NeverReinvocationPolicy),
+					Expression: `
+						Object{
+							metadata: Object.metadata{
+								annotations: {
+									"foo2": "foo2Value"
+								}
+							}
+						}`,
+				}),
+			},
+			bindings: []*v1alpha1.MutatingAdmissionPolicyBinding{
+				binding("policy-1", nil, nil),
+				binding("policy-2", nil, nil),
+			},
+			requestOperation: admissionregistrationv1.Create,
+			requestResource:  corev1.SchemeGroupVersion.WithResource("configmaps"),
+			requestObject: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-configmap",
+					Namespace: "default",
+				},
+			},
+			expected: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-configmap",
+					Namespace: "default",
+					Annotations: map[string]string{
+						"foo1": "foo1Value",
+						"foo2": "foo2Value",
+					},
+				},
+			},
+		},
+		{
+			name: "policy with native param",
+			policies: []*v1alpha1.MutatingAdmissionPolicy{
+				policy("policy-with-param", matchAllResources, &v1alpha1.ParamKind{
+					APIVersion: "v1",
+					Kind:       "ConfigMap",
+				}, v1alpha1.Mutation{
+					PatchType:          ptr.To(v1alpha1.ApplyConfigurationPatchType),
+					ReinvocationPolicy: ptr.To(v1alpha1.NeverReinvocationPolicy),
+					Expression: `
+						Object{
+							metadata: Object.metadata{
+								annotations: {
+									params.data["key"]: params.data["value"]
+								}
+							}
+						}`,
+				}),
+			},
+			bindings: []*v1alpha1.MutatingAdmissionPolicyBinding{
+				binding("policy-with-param", &v1alpha1.ParamRef{
+					Name: "test-param",
+				}, nil),
+			},
+			params: []runtime.Object{
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-param",
+						Namespace: "default",
+					},
+					Data: map[string]string{
+						"key":   "myFooKey",
+						"value": "myFooValue",
+					},
+				},
+			},
+			requestOperation: admissionregistrationv1.Create,
+			requestResource:  corev1.SchemeGroupVersion.WithResource("configmaps"),
+			requestObject: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "request-configmap",
+					Namespace: "default",
+				},
+			},
+			expected: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "request-configmap",
+					Namespace: "default",
+					Annotations: map[string]string{
+						"myFooKey": "myFooValue",
+					},
+				},
+			},
+		},
+		// {
+		// Can't be reliably tested due to large possibly 30s delay in picking up
+		// CRDs in discovery
+		// 	name: "policy with crd param",
+		// },
```

Comment:
@Jefftree Is there anything we can do here?

Reply:
Is there any issue with waiting for up to that amount of time though? It's generally pretty fast <1s and in our discovery tests we also wait up to 30s for discovery to appear and we haven't seen timeout failures https://github.com/kubernetes/kubernetes/blob/master/test/integration/apiserver/discovery/discovery_test.go#L389-L413

Reply:
@cici37 or @alexzielenski can either of you clarify the "Can't be reliably tested" comment here. I'm having trouble reconciling it with what @Jefftree is saying here.